pysat/tests/test_files.py

```diff
@@ -52,14 +45,10 @@ def remove_files(inst=None):
 
 
 # create year doy file set
-def create_files(inst, start=None, stop=None, freq='1D', use_doy=True,
+def create_files(inst, start=pysat.datetime(2009, 1, 1),
+                 stop=pysat.datetime(2013, 12, 31), freq='1D', use_doy=True,
```

Comment:
I believe string defaults a frowned upon. Things can get messed up there. This is the source of my preference for setting optionals to None, though I tend to use it for more than strictly required.
Default keywords should not be set to strings, lists, dicts, anything that can be modified in memory as those changes persist between calls. The method headers are only ever set once in python.

Reply:
I started the string default, it is my bad, but now I see it.

Reply:
Is my use of the start and stop dates ok? It still looks a little odd to me, but it does reduce the uncovered lines. Another option is to remove start and stop defaults, since these are always specified in the testing routine.

Reply:
Replaced with `freq=None`, then added a check to reset within the function.

Also, removed default values for start and stop.  The rationale behind eliminating the `start=None` keywords is that start always has a date specified within the test area, and lines after the `if start is None:` checks show up as missed lines in the code.